File: P10/FCFS_Disk.java
package P10;

import java.util.Scanner;

public class FCFS_Disk {
    public static void main(String[] args) {
       
        Scanner sc = new Scanner(System.in);
        int[] t = new int[20];
        int n, tot = 0;
        int[] tohm = new int[20];
        float avhm;
        System.out.println("enter the no.of tracks");
        n = sc.nextInt();
        System.out.println("enter the tracks to be traversed");
        for (int i = 2; i < n + 2; i++) {
            t[i] = sc.nextInt();
        }
        for (int i = 1; i < n + 1; i++) {
            tohm[i] = Math.abs(t[i + 1] - t[i]);
            // if (tohm[i] < 0) {
            //     tohm[i] = tohm[i] * (-1);
            // }
        }
        for (int i = 1; i < n + 1; i++) {
            tot += tohm[i];
        }
        avhm = (float) tot / n;
        System.out.println("Tracks traversed\tDifference between tracks");
        for (int i = 1; i < n + 1; i++) {
            System.out.printf("%d\t\t\t%d\n", t[i], tohm[i]);
        }
        System.out.printf("\nAverage header movements:%f", avhm);
        sc.close();
    }
}

File: P10/Python P10/FCFS Disk.py
def fcfs_disk():
    t = [0] * 20
    tohm = [0] * 20

    n = int(input("Enter the no. of tracks: "))
    print("Enter the tracks to be traversed:")
    for i in range(2, n + 2):
        t[i] = int(input())

    tot = 0
    for i in range(1, n + 1):
        tohm[i] = abs(t[i + 1] - t[i])
        tot += tohm[i]

    avhm = tot / n
    print("Tracks traversed\tDifference between tracks")
    for i in range(1, n + 1):
        print(f"{t[i]}\t\t\t{tohm[i]}")

    print(f"\nAverage header movements: {avhm:.2f}")


if __name__ == "__main__":
    fcfs_disk()
